lib/rules.js

```diff
@@ -184,6 +185,234 @@ export const MULTIPLY_BY_INVERSE = defineRuleString('#a / (#b / #c)', '#a * (#c
+        const radicand = node.args[0]
+        const variable = radicand.args[0]
+        const exponent = radicand.args[1]
+        let index
+
+        // checks whether index is provided
+        // if not assume square root
+        // e.g. nthRoot(x^2)
+        if (node.args[1]) {
+            index = node.args[1]
+        } else {
+            index = build.numberNode(2)
+        }
+
+        let result
+
+        // simplify exponent / index
+        // e.g. nthRoot(x^2, 4) -> 2/4 -> 1/2
+        const newRoot = applyRule(SIMPLIFY_FRACTION,
+                                  build.applyNode('div', [exponent, index]))
+        let numerator = newRoot.args[0]
+        let denominator = newRoot.args[1]
+
+        // Case #1: numerator > denominator && denominator = 1
+        // e.g nthRoot(x^4, 2) -> 4/2 -> 2/1 -> x^2
+        // Case #2: numerator < denominator ||
+        // numerator > denominator && denominator != 1, return nthRoot
+        // e.g nthRoot(x^2, 4) -> 2/4 -> 1/2 -> nthRoot(x^1, 2)
+        // Case #3: numerator == denominator
+        // e.g nthRoot(x^2, 2) -> 2/2 -> x^1
+
+        if (query.getValue(numerator) > query.getValue(denominator)
+            &&  query.getValue(denominator) == 1){
+            result = build.applyNode('pow', [variable, numerator])
+        } else if (query.getValue(numerator) === query.getValue(denominator)){
+            result = build.applyNode('pow', [variable, build.numberNode(1)])
+        } else {
+            result = build.applyNode (
+                {type: 'Identifier', name: 'nthRoot'},
+                [build.applyNode('pow', [variable, numerator]), denominator]
+            )
+        }
+        return result
+    }
+)
+
+// e.g. nthRoot(2, 2) * nthRoot(3, 2) -> nthRoot(2 * 3, 2)
+export const COMBINE_UNDER_ROOT = defineRuleString('nthRoot(#a, #c) * nthRoot(#b, #c)', 'nthRoot(#a * #b, #c)')
+
+// e.g. 2 * 2 * 2 -> 2 ^ 3
+export const CONVERT_MULTIPLICATION_TO_EXPONENT = defineRuleString('#a^#b_0 * ...', '#a^#eval(#b_0 + ...)', {a: query.isNumber, b: query.isNumber})
+
+// e.g. nthRoot(2 * x) -> nthRoot(2) * nthRoot(x)
+export const DISTRIBUTE_NTH_ROOT = defineRule(
+    (node) => {
+        let isNthRoot = false
+        let validRadicand = false
+        if (query.isApply(node)){
+            isNthRoot = query.isIdentifier(node.op)
+                && node.op.name == 'nthRoot'
+            validRadicand = query.isMul(node.args[0])
+        }
+        return (isNthRoot && validRadicand) ? {node} : null
+    },
+
+    (node) => {
+        const radicand = node.args[0]
+        const result = build.applyNode(
+            'mul',
+            radicand.args.map(arg => {
+                return build.applyNode(
+                    {type: 'Identifier', name: 'nthRoot'},
+                    [arg]
+                )
+            })
+        )
+        return result
+    }
+)
+
+// e.g. nthRoot(4) * nthRoot(x^2) -> 2 * x
+export const EVALUATE_DISTRIBUTED_NTH_ROOT = defineRule(
```

Comment:
If we solve https://github.com/semantic-math/math-rules/issues/23 then rules like this should be trivial.  It'd be great to solve it in a general way so that we don't have to write custom functions each time we want to apply the same rule to multiple nodes.  In this case we're applying one rule or another so we'd nee to https://github.com/semantic-math/math-rules/issues/10 as well.